common/cache/lru_test.go

```diff
@@ -671,3 +671,56 @@ func TestCache_PutIfNotExistWithSameKeys_Pin(t *testing.T) {
 	assert.Equal(t, &testEntryWithCacheSize{3}, val)
 	assert.Equal(t, 3, cache.Size())
 }
+
+func TestCache_ItemSizeChangeBeforeRelease(t *testing.T) {
```

Comment:
Can you break this test apart into multiple tests that check specific behaviors rather than one test for all behaviors? For example:

- `TestCache_EnforcesMaxSizeLimitOnRelease`
- `TestCache_RejectsRequestWhenAtLimit`

etc.

Reply:
I have the test into two tests. But still, one test is too big. It is hard to split it into two. I have added a comment to explain what it does.